### PR TITLE
Update system naming and references

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

If merged, this PR will update naming instances of the system as well as introduce consistency around grammar when the system or team is referenced.

### :hammer_and_wrench: Detailed description

Key content updates:

- **HashiCorp Design System** becomes **Helios Design System**, or in repetitive circumstances, just **Helios**
- **Design System Team** replaces all references to our team
- **design system** replaces all smaller/more repetitive references to the system (i.e. "Design System")
- There were a few instances of HDS that I removed (namely in the `<title>` elements

These changes mainly achieve consistency in the prose.

### :link: External links

Jira ticket: [HDS-1300](https://hashicorp.atlassian.net/browse/HDS-1300)

### CODE REVIEW

File: website/app/components/dynamic-template-error.hbs [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-  <p class="doc-text-body-small">There was an error rendering the dynamic template for this page.<br />Check the console
+  <p class="doc-text-body-small"><strong>Error rendering the dynamic
+      template</strong></p>
+  <p class="doc-text-body-small">There was an error rendering the dynamic
+    template for this page.<br />Check the console
     to see the details of the error.</p>
-  <p class="doc-text-body-small">If you need support or want to notify the design system team of this error, please
-    <LinkTo @route="support" class="doc-link-generic">contact us</LinkTo>.</p>
+  <p class="doc-text-body-small">If you need support or want to notify the
+    Design System Team of this error, please
```
Comment (Dhaulagiri) [inline website/app/components/dynamic-template-error.hbs:8]:
```suggestion
    Design Systems Team of this error, please
```

File: website/app/templates/index.hbs [hunk 2]
```diff
@@ -1,40 +1,29 @@
...
-          @model="getting-started/for-engineers"
-          @label="Install HDS in your project"
-          @icon="arrow-right"
-          @isAnimated={{true}}
-        />
+        <Doc::LinkWithIcon @route="show" @model="getting-started/for-designers"
+          @label="Start designing with HDS" @icon="arrow-right"
+          @isAnimated={{true}} />
+        <Doc::LinkWithIcon @route="show" @model="getting-started/for-engineers"
+          @label="Install HDS in your project" @icon="arrow-right"
```
Comment (Dhaulagiri) [inline website/app/templates/index.hbs:13]:
```suggestion
          @label="Install Helios in your project" @icon="arrow-right"
```

File: website/app/templates/index.hbs [hunk 3]
```diff
@@ -1,40 +1,29 @@
...
-        />
-        <Doc::LinkWithIcon
-          @route="show"
-          @model="getting-started/for-engineers"
-          @label="Install HDS in your project"
-          @icon="arrow-right"
-          @isAnimated={{true}}
-        />
+        <Doc::LinkWithIcon @route="show" @model="getting-started/for-designers"
+          @label="Start designing with HDS" @icon="arrow-right"
```
Comment (Dhaulagiri) [inline website/app/templates/index.hbs:10]:
```suggestion
          @label="Start designing with Helios" @icon="arrow-right"
```

**Review by Dhaulagiri — APPROVED** [review]
👏  left a few suggestions

File: website/docs/about/accessibility-statement.md [hunk 4]
```diff
@@ -47,14 +47,14 @@ If we are unable to provide conformance through the component itself, we will pr
...
-The HashiCorp Design Systems Team self-assesses the accessibility of the HashiCorp Design System by the following approaches:
+The Design Systems Team self-assesses the accessibility of Helios by the following approaches:
 
 - Manual testing with assistive technologies
 - Use of available automated checks
 
 ## Feedback
 
-We welcome your feedback on the accessibility of the HashiCorp Design System. Please let us know if you encounter accessibility barriers on the HashiCorp Design System:
+We welcome your feedback on the accessibility of Helios, please let us know if you encounter accessibility barriers in the design system:
```
Comment (heatherlarsen) [inline website/docs/about/accessibility-statement.md:57]: Grammatically, the period was more accurate, as this updated version turns this into a run-on sentence. Alternatively, you can use a semi-colon. 

```suggestion
We welcome your feedback on the accessibility of Helios. Please let us know if you encounter accessibility barriers in the design system:
```

**Review by heatherlarsen — APPROVED** [review]
Just one small grammar nitpick.

File: website/docs/getting-started/for-designers/partials/best-practices.md [hunk 5]
```diff
@@ -2,7 +2,7 @@ Figma is an incredibly versatile tool, but this versatility can sometimes backfi
 
 !!! Info
 
-These best practices have been curated by the design system team, Figma has it’s own set of [best practices](https://www.figma.com/best-practices/) for using the tool effectively.
+These best practices have been curated by the Design Systems Team, Figma has it’s own set of [best practices](https://www.figma.com/best-practices/) for using the tool effectively.
```
Comment (KristinLBradley) [inline website/docs/getting-started/for-designers/partials/best-practices.md:5]:
```suggestion
These best practices have been curated by the Design Systems Team, Figma has its own set of [best practices](https://www.figma.com/best-practices/) for using the tool effectively.
```

File: website/app/templates/error.hbs [hunk 6]
```diff
@@ -2,23 +2,25 @@
 
 <div class="doc-page-error">
   <div class="doc-page-error__wrapper">
-    <img
-      class="doc-page-error__image"
-      src="/assets/other/jory-travolta.gif"
-      alt="Animated gif created by Jory Tindall, showing a classic John Travolta meme on top of a composition of HDS component"
-    />
+    <img class="doc-page-error__image" src="/assets/other/jory-travolta.gif"
+      alt="Animated gif created by Jory Tindall, showing a classic John Travolta meme on top of a composition of Helios component" />
```
Comment (KristinLBradley) [inline website/app/templates/error.hbs]:
```suggestion
      alt="Animated gif created by Jory Tindall, showing a classic John Travolta meme on top of a composition of Helios components" />
```

**Review by KristinLBradley — APPROVED** [review]
Left just a couple spelling/punctuation fixes

**Pull request merged**